### DISCUSSION TITLE
failed to compute cache key: "/keycloak-horreum.json"

### DISCUSSION THREAD

**Discussion by shivarm** [Hyperfoil/Horreum#414] [General]
I got this error during set up this project locally I run following  docker command.

`docker-compose -p horreum -f infra/docker-compose.yml up -d`

quay.io/keycloak/keycloak:20.0.1@sha256:8830f76112b62231f17b43f115a3deca4de7e4a33eeee298e85cb295c5810e8a                                      0.0s 

 => ERROR [2/4] COPY keycloak-horreum.json /opt/keycloak/data/import/                                                                                        0.0s

------
 > [2/4] COPY keycloak-horreum.json /opt/keycloak/data/import/:
------

failed to compute cache key: "/keycloak-horreum.json" not found: not found
ERROR: Service 'keycloak' failed to build : Build failed

**Comment by jesperpedersen** [comment]
What does `find . -name keycloak-horreum.json` show ?

**Reply by shivarm** [reply]
This above error is comming when i try to pull images.Reffers following image.

![unnamed](https://user-images.githubusercontent.com/91419219/224690968-be626b9f-b1f4-4e98-8498-c0e17f8e34a9.png)

**Reply by shivarm** [reply]
I think it's mean  /keycloak-horreum.json /opt/keycloak/data/import/: is missing

**Reply by jesperpedersen** [reply]
I can recreate this locally

**Reply by shivarm** [reply]
Thanks

**Reply by shivarm** [reply]
I hope this would be fixing soon.

**Reply by johnaohara** [reply]
I have a fix i will open a PR now

**Reply by johnaohara** [reply]
https://github.com/Hyperfoil/Horreum/pull/417

**Comment by jesperpedersen** [comment]
This should be resolved now.

**Reply by shivarm** [reply]
Yes thanks now it working fine @johnaohara @jesperpedersen

**Comment by shivarm** [comment]
# This time i got new error

![image](https://user-images.githubusercontent.com/91419219/224934945-d6c8d35d-f3a0-45dd-95e6-e9659c886875.png)

**Reply by shivarm** [reply]
```
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager]  Failed to compile.
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager]
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] ./src/domain/tests/Test.tsx
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] Attempted import error: './Actions' does not contain a default export (imported as 'Actions').   
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager]
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager]
js" run build
[ERROR]         at io.quarkiverse.quinoa.deployment.packagemanager.PackageManager.build(PackageManager.java:56)
[ERROR]         at io.quarkiverse.quinoa.deployment.QuinoaProcessor.processBuild(QuinoaProcessor.java:138)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:829)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR][ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :horreum-backend ```

**Reply by jesperpedersen** [reply]
Did you use

```
export NODE_OPTIONS=--openssl-legacy-provider
mvn clean package
mvn -Dquarkus.test.continuous-testing=disabled quarkus:dev
```

?

**Reply by shivarm** [reply]
yes i used 

```
export NODE_OPTIONS=--openssl-legacy-provider
mvn -DskipTests=true package
mvn -Dquarkus.test.continuous-testing=disabled quarkus:dev
```

**Reply by shivarm** [reply]
now i am  going to try yours

```
export NODE_OPTIONS=--openssl-legacy-provider
mvn clean package
mvn -Dquarkus.test.continuous-testing=disabled quarkus:dev
```

**Reply by shivarm** [reply]
same error got https://github.com/Hyperfoil/Horreum/discussions/414#discussioncomment-5300269

**Reply by johnaohara** [reply]
@shivam-sharma7 which branch are you using?

**Reply by johnaohara** [reply]
This is an error attempting to build the UI

**Reply by jesperpedersen** [reply]
What is the system ? What version of Java ? What version of Maven ?

**Reply by jesperpedersen** [reply]
Can you try `rm -Rf webapp/node*` and then the whole thing again ?

**Reply by shivarm** [reply]
> @shivam-sharma7 which branch are you using?

master

**Reply by shivarm** [reply]
> What is the system ? What version of Java ? What version of Maven ?

Ubuntu WSL, openjdk 11.0.18, Apache Maven 3.6.3

**Reply by shivarm** [reply]
> Can you try `rm -Rf webapp/node*` and then the whole thing again ?

okay

**Reply by shivarm** [reply]
This time run single `mvn clean package` then also i got same error.
